cmd/internal/cli/push.go

```diff
@@ -37,6 +48,34 @@ var PushCmd = &cobra.Command{
 
 		// Push to library requires a valid authToken
 		if authToken != "" {
+			if _, err := os.Stat(args[0]); os.IsNotExist(err) {
+				sylog.Fatalf("Unable to open: %v: %v", args[0], err)
+			}
+			if !unauthenticatedPush {
+				// check if the container is signed
+				imageSigned, err := signing.IsSigned(args[0], "https://keys.sylabs.io", 0, false, authToken, true)
+				if err != nil {
+					// err will be: "unable to verify container: %v", err
+					sylog.Warningf("%v", err)
+				}
+				// if its not signed, print a warning
+				if !imageSigned {
+					sylog.Infof("TIP: Learn how to sign your own containers here : https://www.sylabs.io/docs/")
+					fmt.Fprintf(os.Stderr, "\n")
+					fmt.Printf("Your container is **NOT** signed! You REALLY should sign your container before pushing!\n")
+					reps, err := sypgp.AskQuestion("Do you really want to continue? [N/y] ")
```

Comment:
no questions, the code got here because the user did not explicitly request to skip verification. If they want to push unauthenticated containers, they MUST pass --allow-unauthenticated (which also makes me think that in this specific case "unauthenticated" might be the wrong term, as there's no authentication going on, as it's only checking that the image has a signature and that that signature can be verified, maybe the option should be called "--allow-unsigned")